pkg/template/jsonpath_test.go

```diff
@@ -0,0 +1,145 @@
+package template_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/triggers/pkg/template"
+)
+
+var objects = `{"a":"v","c":{"d":"e"},"empty": "","null": null, "number": 42}`
+var arrays = `[{"a": "b"}, {"c": "d"}, {"e": "f"}]`
+
+// Checks that we print JSON strings when the JSONPath selects
+// an array or map value and regular values otherwise
+func TestParseJSONPath(t *testing.T) {
+	var objectBody = fmt.Sprintf(`{"body":%s}`, objects)
+	tests := []struct {
+		name string
+		expr string
+		in   string
+		want string
+	}{{
+		name: "objects",
+		in:   objectBody,
+		expr: "$(body)",
+		// TODO: Do we need to escape backslashes for backwards compat?
```

Comment:
@ncskier @vtereso In the current gjson bindings, we always escape the quotes `\"` in the returned value. Why do we do that?

Reply:
I think it's because we're passing we're passing that JSON object as a type String in the ResourceTemplate JSON: https://github.com/tektoncd/triggers/blob/master/pkg/sink/sink.go#L195

Reply:
ok, I'm adding escaping behavior for now for backwards compatibility. I'll open an issue to explore alternative approaches here -- maybe smarter concatenation somehow or using json patches

Reply:
Sounds good, thanks!